Tests/ApolloCodegenTests/GlobTests.swift

```diff
@@ -0,0 +1,359 @@
+import XCTest
+import Nimble
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+class GlobTests: XCTestCase {
+  let baseURL = CodegenTestHelper.outputFolderURL().appendingPathComponent("Glob")
+
+  // MARK: Setup
+
+  override func setUpWithError() throws {
+    let fileManager = FileManager.default.apollo
+
+    // <outputFolder>/Glob/
+    try fileManager.createDirectoryIfNeeded(atPath: baseURL.path)
+
+    let files = [
+      self.baseURL.appendingPathComponent("file.one").path, // <outputFolder>/Glob/file.one
+      self.baseURL.appendingPathComponent("file.two").path, // <outputFolder>/Glob/file.two
+      self.baseURL.appendingPathComponent("a/file.one").path, // <outputFolder>/Glob/a/file.one
+      self.baseURL.appendingPathComponent("a/b/file.one").path, // <outputFolder>/Glob/a/b/file.one
+      self.baseURL.appendingPathComponent("a/b/c/file.one").path, // <outputFolder>/Glob/a/b/c/file.one
+      self.baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path, // <outputFolder>/Glob/a/b/c/d/e/f/file.one
+      self.baseURL.appendingPathComponent("a/b/c/d/e/f/file.two").path, // <outputFolder>/Glob/a/b/c/d/e/f/file.two
+      self.baseURL.appendingPathComponent("other/file.one").path, // <outputFolder>/Glob/other/file.one
+      self.baseURL.appendingPathComponent("other/file.oye").path // <outputFolder>/Glob/other/file.oye
+    ]
+
+    for file in files {
+      expect(try fileManager.createFile(atPath: file)).to(beTrue())
+    }
+  }
+
+  override func tearDownWithError() throws {
+    try FileManager.default.apollo.deleteDirectory(atPath: baseURL.path)
+  }
+
+  // MARK: Tests
+
+  func test_match_givenSinglePattern_whenNoMatch_shouldReturnEmpty() throws {
+    // given
+    let pattern = baseURL.appendingPathComponent("*.xyz").path
+
+    // when
+    // <baseURL>/file.one
+    // <baseURL>/file.two
```

Comment:
For clarity: These are all just comments because `files` are created in the setup function, correct? 

The one thing I'd be a little worried about with this method is that someone could change what's in `files` but not update these comments and then you'd get some nonsensical results

Reply:
Yes - these are comments to easier see what files are present at the test paths. I agree they could get easily become stale. I'm happy to remove them if they've got little value; maybe the value was more for me as I was building the tests? 🤔

Reply:
With 5db89203 the file creation is now part of each test. Cleans up a couple things; no more comments and only create what you need to satisfy the test. Overall a good change. 🙇🏻

Reply:
I think it actually might be worth it to have all the files created in one place and have them be the same set of files for each test, especially if what you're trying to test is that certain things are *not* being found - but maybe just make the `createFiles` call one in each test in the `when` rather than the setup?

Reply:
I prefer having duplication and each test case able to be read independently. If you're not strongly opposed I'm going to leave the files array in each test case to prevent having to refer outside the test to know what files were created. I did expand on which paths are created to cover the edges though - https://github.com/apollographql/apollo-ios/pull/2041/commits/33b961e9cbe7377b9257db6f523758babd441e1f. I'm confident in the tests.

Reply:
Nope, not strongly opposed, fine by me.